sycl/doc/extensions/DeviceIf/device_if.asciidoc

```diff
@@ -0,0 +1,391 @@
+= SYCL_EXT_ONEAPI_DEVICE_IF
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+Copyright (c) 2021-2021 Intel Corporation.  All rights reserved.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+This extension is written against the SYCL 2020 revision 3 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Introduction
+
+This extension provides a way for device code to query the device on which it
+is running in order to conditionally use features that may not be supported on
+all devices.  This is different from the existing `device::has()` function
+because the extension can be called from device code while `device::has()` can
+only be called from host code.
+
+The motivating use case for this extension is for developers who provide device
+side libraries.  To illustrate, consider an application developer (i.e. someone
+who is **not** developing a device side library) who wants to code a kernel
```

Comment:
Nit: I think "device-side library" is easier to read here (and same comment for other usages of "device side library").

Reply:
Thanks, makes sense.